src/NUnitFramework/framework/Constraints/DefaultConstraint.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// Tests that the actual value is default value for type.
+    /// </summary>
+    public class DefaultConstraint : Constraint
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultConstraint"/> class.
+        /// </summary>
+        public DefaultConstraint()
+        {
+            this.Description = "default";
+        }
+
+        /// <summary>
+        /// Applies the constraint to an actual value, returning a ConstraintResult.
+        /// </summary>
+        public override ConstraintResult ApplyTo<TActual>(TActual actual)
+        {
+            var actualType = actual?.GetType();
+
```

Comment:
What is the reason for not using `typeof(TActual)`?
Is this because a lot of thing in NUnit get cast to `object` and `TActual` is not relevant?
The problem this leads to is that `int?` v = 0 has an actual type of `int` and returns true for `Is.Default`

Reply:
You're right, I decided simply to use default(TActual) after discussion in original issue.